### PR TITLE
Milestone 2: Decorate a class

### PR DESCRIPTION
**Pull request by Sahar-AbdelSamad**
## OOP School LIbrary milestone 2: Decorate a class
In this PR I:
- Created a class `Nameable`and implemented a method called `correct_name` that will raise a `NotImplementedError`
- Added method `correct_name` that returns the name attribute, in `Person` class that inherits from `Nameable`
- Created `Decorator` class that inherits from `Namable` and Implemented `correct_name` method that returns the result of the `correct_name` method of the @nameable
- Created `CapitalizeDecorator` and `TrimmerDecorator` classes that inherit from the `Decorator` class
- Implemented method `correct_name` for `CapitalizeDecorator`  that capitalizes the output of @nameable.correct_name
- Implemented method `correct_name` for `TrimmerDecorator` hat makes sure that the output of @nameable.correct_name has a maximum of 10 characters. If it's longer it should trim the word

### CODE REVIEW

File: decorator.rb [hunk 1]
```diff
@@ -0,0 +1,24 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
```
Comment (bimbolabuari) [inline decorator.rb]: I suggest you add  `nameable` as an attribute accessor in the `Decorator` class 
    `attr_accessor :nameable`

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @Sahar-AbdelSamad, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights
- Create a class Nameable.
- Implement a method called correct_name that will raise a NotImplementedError.
- Prepare base Decorator
 - Prepare CapitalizeDecorator and TrimmerDecorator


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: decorator.rb (hunk 1 above)
Comment (Sahar-AbdelSamad) [reply decorator.rb]: Thank you :smile: I implemented it :+1:

**Review by Cathella — APPROVED** [review]
## 🔥 Status: Approved 🟢

### Hi @Sahar-AbdelSamad 👋🏻, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

rosycheeks-mochi-peach


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**